go/test/e2e/invoke_api_test.go

```diff
@@ -298,3 +298,61 @@ func TestInvokeExternalAgent(t *testing.T) {
 		require.Contains(t, text, "kebab for user@example.com", string(jsn))
 	})
 }
+
+func TestInvokeDeclarativeAgentWithMcpServerTool(t *testing.T) {
```

Comment:
Rather than spinning up these agents as a part of the CI job, can we spin up the agent + mcp server at the beginning of the test? That's also how the above: `TestInvokeInlineAgent` works

Reply:
done. Refactored the setup functions to make them more reusable while I was at it.